### PR TITLE
Adding vertical neutral-neutral friction

### PR DESCRIPTION
**Pull request by roober-t**
# Description

Addresses #80

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change that adds functionality)

# How Has This Been Tested?

I added output (with verbose level 4) that prints each coefficient from the vector.

## Test configuration

* Operating system: macOS Ventura 13.1
* Compiler, version number: gcc Apple clang version 14.0.0

# Checklist:

- [N/A] Make sure you are merging into the ``develop`` (not ``master``) branch
- [X ] My code follows the style guidelines of this project
- [X ] I have performed a self-review of my own code
- [X ] I have commented my code, particularly in hard-to-understand areas
- [N/A] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [N/A] New and existing unit tests pass locally with my changes
- [N/A] Any dependent changes have been merged and published in downstream modules
- [N/A] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by aburrell** [comment]
@roober-t can you rename this pull request to reflect the changes instead of calling it "develop"? Perhaps something like "Adding vertical neutral-neutral friction"?

**Review by aburrell — COMMENTED** [review]
An initial review for style expectations for the develop branch.  I also recommend referring to the developers guide for commit message style to ensure future commits are descriptive.

**Comment by roober-t** [comment]
Since the initial pull request, I have added a newline to the bottom of ion_neutral_collision_frequencies. I also got rid of all extraneous comments throughout the code, including a few print statements in read_collision_file.cpp.

**Comment by aaronjridley** [comment]
@aburrell - can you do a quick check of this?  I have cleaned it all up, made a test, and verified that everything works (found a major bug in the vertical gradient for non-uniform grids!).  This should work well with the vertical advection pull request that is pending.

File: include/aether.h [hunk 1]
```diff
@@ -130,8 +130,10 @@ using json = nlohmann::json;
 // not done
 #include "parallel.h"
 
+// not done
```
Comment (aburrell) [inline include/aether.h:133]: Why is this commented "not done"?

File: src/calc_euv.cpp [hunk 2]
```diff
@@ -29,11 +29,10 @@ int calc_euv(Planets planet,
     std::string function = "Euv::calc_euv";
     static int iFunction = -1;
     report.enter(function, iFunction);
-
+
```
Comment (aburrell) [inline src/calc_euv.cpp:32]:
```

File: src/calc_neutral_derived.cpp [hunk 3]
```diff
@@ -7,34 +7,165 @@
...
+//  Calculate mass density and number density:
+// ----------------------------------------------------------------------
+
+void Neutrals::calc_mass_density(Report &report) {
+    
   std::string function = "Neutrals::calc_mass_density";
   static int iFunction = -1;
   report.enter(function, iFunction);
-
+
```
Comment (aburrell) [inline src/calc_neutral_derived.cpp:42]:
```

File: src/calc_neutral_derived.cpp [hunk 4]
```diff
@@ -7,34 +7,165 @@
...
   std::string function = "Neutrals::calc_mass_density";
   static int iFunction = -1;
   report.enter(function, iFunction);
-
+    
   rho_scgc.zeros();
   density_scgc.zeros();
-
-  for (iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
+
```
Comment (aburrell) [inline src/calc_neutral_derived.cpp:45]:
```suggestion

```

File: src/calc_neutral_derived.cpp [hunk 5]
```diff
@@ -7,34 +7,165 @@
...
-
-  for (iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
+    
+  for (int64_t iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
     rho_scgc = rho_scgc +
-               species[iSpecies].mass * species[iSpecies].density_scgc;
+      species[iSpecies].mass * species[iSpecies].density_scgc;
     density_scgc = density_scgc + species[iSpecies].density_scgc;
   }
+  report.exit(function);
```
Comment (aburrell) [inline src/calc_neutral_derived.cpp:51]: Missing return at end of function here and in other functions in this file.

File: src/init_geo_grid.cpp [hunk 6]
```diff
@@ -615,7 +615,7 @@ bool Grid::init_geo_grid(Quadtree quadtree,
   // Calculate magnetic field and magnetic coordinates:
   fill_grid_bfield(planet, input, report);
 
-  // Throw a little message for students:
+  //Throw a little message for students:
```
Comment (aburrell) [inline src/init_geo_grid.cpp:618]:
```suggestion
  // Throw a little message for students:
```

File: src/neutral_ion_collisions.cpp [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+	  rho_i % ions.species[iIon].nu_ion_neutral_vcgc[iNeutral] %
+	  (ions.species[iIon].par_velocity_vcgc[iDir] +
+	   ions.species[iIon].perp_velocity_vcgc[iDir] -
+	   neutrals.species[iNeutral].velocity_vcgc[iDir]);
+      } // for each ion
+      neutrals.species[iNeutral].acc_ion_drag[iDir] = rho_sum / rho_n;
+    } // for each direction
+  } // for each neutral
+    
+  report.exit(function);
```
Comment (aburrell) [inline src/neutral_ion_collisions.cpp:41]: Missing return

File: src/neutrals.cpp [hunk 8]
```diff
@@ -172,20 +181,13 @@ void Neutrals::fill_with_hydrostatic(Grid grid, Report report) {
...
-    // Integrate with hydrostatic equilibrium up:
-    species[iSpecies].scale_height_scgc =
-      cKB * temperature_scgc /
-      (species[iSpecies].mass * abs(grid.gravity_vcgc[2]));
-
     for (int iAlt = 1; iAlt < nAlts; iAlt++) {
       species[iSpecies].density_scgc.slice(iAlt) =
         species[iSpecies].density_scgc.slice(iAlt - 1) %
         exp(-grid.dalt_lower_scgc.slice(iAlt) /
             species[iSpecies].scale_height_scgc.slice(iAlt));
```
Comment (aburrell) [inline src/neutrals.cpp:188]: Now that the lines that defined `species[iSpecies].scale_height_scgc` have been removed, where is this set?

File: src/read_collision_file.cpp [hunk 9]
```diff
@@ -425,4 +453,162 @@ void parse_bst_in_table(std::vector<std::vector<std::string>> csv,
...
+  if (report.test_verbose(4)) {
+    for (int i = 0; i < neutrals.nSpecies; i++) {
+      cout << "Current species: " << neutrals.species[i].cName << endl;
+      for (int j = 0; j < neutrals.species[i].diff0.size(); j++)
+        cout << neutrals.species[i].diff0[j] << ", ";
+      cout << endl;
+    }
+  }
+
+  report.exit(function);
```
Comment (aburrell) [inline src/read_collision_file.cpp:535]: Missing return for these functions.

File: tests/eddy/run_test.sh [hunk 10]
```diff
@@ -0,0 +1,77 @@
+#!/bin/sh
```
Comment (aburrell) [inline tests/eddy/run_test.sh:1]: Add a command to run this in the CI.

**Review by aburrell — COMMENTED** [review]
Also need to remove the two `.DS_Store` files that were committed before the `.gitignore` file was updated.

File: include/aether.h (hunk 1 above)
Comment (aaronjridley) [reply include/aether.h:133]: These are notes that are reminders that the commenting of the include file is not done yet.

File: src/neutrals.cpp (hunk 8 above)
Comment (aaronjridley) [reply src/neutrals.cpp:188]: In the calc scale height function.  We needed to do that to incorporate eddy diffusion, which forces all of the scale heights to be the same.

**Comment by aaronjridley** [comment]
Fixed some of the formatting changes that were suggested, merged the changes with the develop branch (removed report and input from the function calls), and fixed some bugs.

**Pull request merged**